client/src/components/helpers/AvatarRenderer.js

```diff
@@ -15,9 +15,25 @@ const propTypes = {
 
 function AvatarRenderer({ picture, userName, isDonating, isTopContributor }) {
   const { t } = useTranslation();
+  const [isPictureValid, setIsPictureValid] = useState(true);
   let borderColor = borderColorPicker(isDonating, isTopContributor);
-  let isPlaceHolderImage =
-    /example.com|identicon.org/.test(picture) || picture === defaultUserImage;
+  const onImageLoad = () => setIsPictureValid(true);
+  const onImageError = () => setIsPictureValid(false);
+  let validationImage = document.createElement('img');
+  validationImage.src = picture;
+  validationImage.onload = onImageLoad;
+  validationImage.onerror = onImageError;
+
+  useEffect(() => {
+    validationImage.onload();
+    validationImage.onerror();
+    // eslint-disable-next-line
```

Comment:
```suggestion
  useEffect(() => {
    const validationImage = document.createElement('img');
    validationImage.src = picture;
    validationImage.onload = onImageLoad;
    validationImage.onerror = onImageError;
    // eslint-disable-next-line
```

Reply:
Maybe something like (_have not tested_):
```javascript
function AvatarRenderer({ picture, userName, isDonating, isTopContributor }) {
  const { t } = useTranslation();
  const [isPictureValid, setIsPictureValid] = useState(true);
  const validationImage = useRef(null);
  let borderColor = borderColorPicker(isDonating, isTopContributor);
  const onImageLoad = () => setIsPictureValid(true);
  const onImageError = () => setIsPictureValid(false);

  useEffect(() => {
    validationImage.current.onload = onImageLoad;
    validationImage.current.onerror = onImageError;
    // eslint-disable-next-line
  }, [picture]);

  const isPlaceHolderImage =
    !isPictureValid ||
    /example.com|identicon.org/.test(picture) ||
    picture === defaultUserImage;

  return (
    <div className={`avatar-container ${borderColor}`}>
      {isPlaceHolderImage ? (
        <DefaultAvatar className='avatar default-avatar' />
      ) : (
        <Image
          ref={validationImage}
          alt={t('profile.avatar', { username: userName })}
          className='avatar'
          responsive={true}
          src={picture}
        />
      )}
    </div>
  );
}
```

Reply:
Not sure about this solution and if the use of useRef is appropriate here.

Reply:
The first suggestion I have tested. The second I have not.